File: Striver/Day 25 : DP-1/Longest Common Seq.cpp
class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {
        int m = text1.length(), n = text2.length();
        int t[m+1][n+1];
        for (int i=0; i<m+1; i++)
            t[i][0] = 0;
        for (int j=0; j<n+1; j++)
            t[0][j] = 0;
        for (int i=1; i<m+1; i++){
            for (int j=1; j<n+1; j++){
                if (text1[i-1] == text2[j-1]){
                    t[i][j] = 1 + t[i-1][j-1];
                }
                else{
                    t[i][j] = max(t[i-1][j] , t[i][j-1]);
                }
            }
        }
        return t[m][n];
    }
};

File: Striver/Day 25 : DP-1/Longest Common Seq.java
class Solution {
    public int longestCommonSubsequence(String text1, String text2) {
        
        int[][] dp = new int[text1.length()+1][text2.length()+1];
        
        
        for(int i=1 ; i<dp.length ;i++)
        {
            for(int j=1 ;j<dp[0].length ; j++)
            {
                if(text1.charAt(i-1) == text2.charAt(j-1))
                    dp[i][j] = 1 + dp[i-1][j-1];
                
                else
                    dp[i][j] = Math.max(dp[i-1][j] , dp[i][j-1]);
            }
        }
        
        return dp[text1.length()][text2.length()];
        
        
    }
}
